### ISSUE TITLE
[css-overflow] Top layer containing block and scrollbar-gutter

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#9904]
It seems a bit weird to me that a test-case like this [live](https://crisal.io/tmp/scrollbar-gutter-top-layer.html):

```html
<!doctype html>
<html style="scrollbar-gutter: stable both-edges;">
<div style="width: 80ch; outline: auto;">
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</div>
<dialog id="dialog">
  Woo
</dialog>
<button onclick="this.requestFullscreen()">Go fullscreen</div>
<button onclick="dialog.showModal()">Show modal dialog</div>
```

Makes the fullscreen element not cover the whole screen / its containing block origin not start at zero, due to the scrollbar gutters.

Maybe the top layer fixed-pos containing block should actually start at zero / ignore scrollbars on the root?

It's a bit weird for regular fixed-pos elements too, for what is worth, but that behavior is consistent with `overflow: scroll` at least.

cc: @fantasai @tabatkins @aethanyc @dholbert @smfr @nt1m @mfreed7

**Comment by kizu** [comment]
Being able to position stuff above the root scrollbar would be great, as otherwise we have to do the following:

1. Read the width of the scrollbar.
2. Apply `overflow: hidden` to the root alongside adding it back via a `padding`, so the elements below don't shift (potentially reading the existing padding value to restore it later).
3. Now we can position something above everything.

So having some way to make an element in the top layer go beyond the default scrollbar will be great, and doing this for fixed position makes sense (as abspos we'd want to be under the default scrollbar).

**Comment by emilio** [comment]
@astearns was this closed on purpose?

**Comment by astearns** [comment]
No, this is the “helpful” new projects interface doing things I didn’t ask it to do. Thanks for notifying me.

**Comment by yisibl** [comment]
> RESOLVED: (top layer or fullscreen) elements with 'position: fixed' cover the scrollbars/gutters

I'm confused by the resolution here, which doesn't solve the problem with the `position: fixed` element.

The `<dialog>` element doesn't create the top layer when `.show()` is called, and when the author can set it to `position: fixed`, the page is offset. 

See demo https://codepen.io/yisi/pen/ExzwMgP?editors=0010

https://github.com/w3c/csswg-drafts/assets/2784308/c3e981a8-6a0a-458d-9499-b3a807ddf5aa

Firefox implements it well and doesn't have problems with offsets.

This has been a long-standing user request, with 21 votes in [Chrome's bug](https://issues.chromium.org/issues/40792788).

**Comment by kizu** [comment]
@yisibl Doing this for _any_ fixed element can result in a loss of the scrollbar functionality, where some pre-existing fixed element could go over the scrollbar, making it inaccessible.

Dialog case looks like close to what we want to achieve with the issue: I agree that we'd want it to be over the scrollbars, but we probably need to handle this specific case separately, as we can't make every fixed element go over the root scrollbar-gutter.

**Comment by jantimon** [comment]
Here is a [Chrome bug](https://issues.chromium.org/issues/40835364) which tries to describe exactly what @kizu said

Adding `scrollbar-gutter: stable;` breaks the scroll-bars:

https://github.com/user-attachments/assets/2c21658e-9f86-427a-a129-9da85a01fe9c

**Comment by yisibl** [comment]
News: [Chrome bug 1251856](https://issues.chromium.org/issues/40792788) is now fixed.

But Chrome and Firefox has another problem: `::backdrop` does not override gutter. See:  https://scrollbar-gutter-stable-bug.glitch.me/

![image](https://github.com/w3c/csswg-drafts/assets/2784308/e675d79f-7ead-4645-b4cb-d60da3db1d87)

**Comment by froschman** [comment]
The red background is caused by "background-color: red" defined in "html:has(dialog[open]:modal) {...}"

**Comment by marto55555** [comment]
I've also reported it to Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=2021874

> Demo: https://scrollbar-gutter-stable-bug.glitch.me/

Because the website is gone, I've recreated the demo: https://scrollbar-gutter-stable-backdrop.pages.dev/
